tron/eventbus.py

```diff
@@ -144,9 +145,18 @@ def sync_load_log(self):
         duration = time.time() - started
         log.info(f"log read from disk, took {duration:.4}s")
 
-    def sync_save_log(self, reason):
+    def sync_save_log(self, reason: str) -> bool:
         started = time.time()
         new_file = os.path.join(self.log_dir, f"{int(started)}.pickle")
+        previous_file: Optional[str] = os.path.realpath(os.path.join(self.log_dir, "current"))
+        # if we're starting  a fresh Tron server, there won't be a current symlink
+        # and the above line will give us the path to what will eventually be the current
+        # symlink...which is undesirable since we clean up whatever this points to :p
+        # we can tell if this is happening since this previous_file variable should
+        # always point to a file that ends in .pickle under normal operation
+        if previous_file and previous_file.endswith("current"):
```

Comment:
is the `previous_file.endswith("current")` part redundant? Since we are defining it in line 151, we can assume it would always end in "current" if it is not None.

Reply:
it would be, but `current` is a symlink and `os.path.realpath` will return the actual path of whatever you give it (so if you give it a normal file/dir, it'll return the same thing - but if you give it a symlink it'll return the symlink target